### DISCUSSION TITLE
How do I get the current route?

### DISCUSSION THREAD

**Discussion by atrauzzi** [remix-run/remix#5297] [Q&A] [answered]
I don't necessarily mean the current _path_.  But the current route object that has been matched.

In particular, I'm using nested routes, so using the current path isn't straightforward as I may only want to be matching a segment, or a segment that contains params.

(surely this state is avaialble somewhere within react router and I can just pluck it out :slightly_smiling_face:  ?)

**Comment by sergiodxa** [comment, marked as answer]
The [`useMatches` hook](https://remix.run/docs/en/v1/hooks/use-matches) let you get the list of currently matching routes.

There's no way to get a single route because a URL match multiple ones, but the array is ordered, so the first match returned is always the root route and the last one is the leaf route, in the middle you will get parent routes.

Since routes are static it should be possible to know that if you want the parent route match you can do:

```ts
let matches = useMatches()
let parentMatch = matches.at(-2)
```

Also, if you only want another route data there's a [`useRouteLoaderData` hook](https://reactrouter.com/en/6.8.0/hooks/use-route-loader-data), you pass the route id and receive the data or undefined.

**Comment by atrauzzi** [comment]
Thanks @sergiodxa, I'll check it out :slightly_smiling_face:
